### PR TITLE
[extractor/anvato] Fix extractor

### PR DESCRIPTION
**Pull request by bashonly**
</details>

Anvato is a video streaming backend service that is utilized by many North American news sites, as well as all of the official NFL sites. The NFL and CBSLocal extractors currently depend on this extractor, and with this extractor fixed, they could also be fixed in separate PRs (or I could add them onto this PR if that is preferred).

This PR:

 - Fixes API endpoints. Parent company Google has rebranded Anvato as `lura.live` and discontinued use of the old `anvato.net` domain. As a result, all of the API endpoints had changed. Despite the change of URLs, all of its internals (API, JS, embed code, HLS manifests) still refer to the service as Anvato. Because of that, and because of the `anvato:` prefix, I think this extractor should continue to be named `anvato`.

 - Fixes HLS extraction, which was broken due to the API returning JSON on the initial request for the HLS manifest, and due to subtitle-only m3u8s being processed as video formats.
 
 - Removes the extraneous token generator submodule. It was essentially a single function, and it has been consolidated into `anvato.py`. I'm assuming whoever originally implemented the submodule thought there were going to be more token generators needed, but the NFL sites seem to be only ones that have their own token generation method.

 - Simplifies some unnecessary complexity around the `_server_time` method

 - Adds token smuggling from embed extraction

 - Replaces old tests with new ones, since the old ones were long gone

 - Updates code to yt-dlp conventions wherever possible

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by pukkandan** [comment]
Remove anvato_token_generator module from makefile and setup.py

File: yt_dlp/extractor/anvato.py [hunk 1]
```diff
@@ -202,86 +244,79 @@ class AnvatoIE(InfoExtractor):
...
+            note='Fetching server time').get('server_time'))
 
-        self.__server_time = int(self._download_json(
-            self._api_prefix(access_key) + 'server_time?anvack=' + access_key, video_id,
-            note='Fetching server time')['server_time'])
+        if not server_time:
+            self._SERVER_TIME = int(time.time())
 
-        return self.__server_time
+        return server_time
```
Comment (pukkandan) [inline yt_dlp/extractor/anvato.py:280]: this is not caching correctly.
```py
    def _server_time(self, access_key, video_id):
        self._SERVER_TIME = self._SERVER_TIME or int_or_none(self._download_json(
            f'{self._API_BASE_URL}/server_time', video_id, query={'anvack': access_key},
            note='Fetching server time', fatal=False).get('server_time')) or int(time.time())

        return self._SERVER_TIME
```

File: yt_dlp/extractor/anvato.py [hunk 2]
```diff
@@ -5,31 +5,73 @@
...
     unsmuggle_url,
 )
 
 
 def md5_text(s):
-    if not isinstance(s, compat_str):
-        s = compat_str(s)
+    if not isinstance(s, str):
+        s = str(s)
     return hashlib.md5(s.encode('utf-8')).hexdigest()
```
Comment (pukkandan) [inline yt_dlp/extractor/anvato.py:28]:
```suggestion
    return hashlib.md5(str(s).encode()).hexdigest()
```

File: yt_dlp/extractor/anvato.py [hunk 3]
```diff
@@ -296,15 +331,32 @@ def _get_anvato_videos(self, access_key, video_id):
                 'tbr': tbr or None,
             }
 
-            if media_format == 'm3u8' and tbr is not None:
+            vtt_subs, hls_subs = {}, {}
+            if media_format == 'vtt':
+                no_fmt, vtt_subs = self._extract_m3u8_formats_and_subtitles(
+                    video_url, video_id, ext='vtt', entry_protocol='m3u8_native',
+                    m3u8_id='vtt', fatal=False)
+                continue
```
Comment (pukkandan) [inline yt_dlp/extractor/anvato.py:339]:
```suggestion
            if media_format == 'vtt':
                _, vtt_subs = self._extract_m3u8_formats_and_subtitles(
                    video_url, video_id, m3u8_id='vtt', fatal=False)
                continue
```

File: yt_dlp/extractor/anvato.py [hunk 4]
```diff
@@ -296,15 +331,32 @@ def _get_anvato_videos(self, access_key, video_id):
...
-                formats.extend(self._extract_m3u8_formats(
-                    video_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id='hls', fatal=False))
+                # For some videos the initial m3u8 URL returns JSON instead
+                try:
+                    manifest_json = self._download_json(
+                        video_url, video_id, note='Downloading manifest JSON')
+                except ExtractorError:
+                    pass
+                else:
```
Comment (pukkandan) [inline yt_dlp/extractor/anvato.py:352]:
```suggestion
                manifest_json = self._download_json(
                    video_url, video_id, note='Downloading manifest JSON', errnote=False)
                if manifest_json:
```

File: yt_dlp/extractor/anvato.py [hunk 5]
```diff
@@ -296,15 +331,32 @@ def _get_anvato_videos(self, access_key, video_id):
...
+                    video_url = manifest_json.get('master_m3u8')
+                    if not video_url:
+                        continue
+                hls_fmts, hls_subs = self._extract_m3u8_formats_and_subtitles(
+                    video_url, video_id, ext='mp4', entry_protocol='m3u8_native',
+                    m3u8_id='hls', fatal=False)
```
Comment (pukkandan) [inline yt_dlp/extractor/anvato.py:358]:
```suggestion
                    video_url, video_id, ext='mp4', m3u8_id='hls', fatal=False)
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by pukkandan** [comment]
>  The NFL and CBSLocal extractors currently depend on this extractor, and with this extractor fixed, they could also be fixed in separate PRs (or I could add them onto this PR if that is preferred).

I think new PR is better

File: yt_dlp/extractor/anvato.py (hunk 1 above)
Comment (bashonly) [reply yt_dlp/extractor/anvato.py:280]: oh whoops I meant to remove it as a class var entirely and missed a spot. I don't think think it should be cached at all, a new timestamp is required for each API call

File: yt_dlp/extractor/anvato.py (hunk 4 above)
Comment (bashonly) [reply yt_dlp/extractor/anvato.py]: should this also be `fatal=False` if there's a chance it will be parsing an actual m3u8 instead of JSON?

Comment (pukkandan) [reply yt_dlp/extractor/anvato.py]: errnote=False implies non fatal

**Pull request merged**